number_theory/numeric.h

```diff
@@ -0,0 +1,48 @@
+#ifndef NUMBER_THEORY_NUMERIC_H
```

Comment:
In order to follow [this](https://google.github.io/styleguide/cppguide.html#The__define_Guard) style, let's put a trailing underscore such as `NUMBER_THEORY_NUMERIC_H_`.